test/deprecations.test.ts

```diff
@@ -0,0 +1,11 @@
+import * as fs from 'node:fs'
+import * as path from 'node:path'
+import {test, expect} from 'vitest'
+import {tsFileErrors} from './ts-output'
+
+test.each(['usage.test.ts', 'types.test.ts'])('%s: toMatchTypeOf matches toExtend behaviour', file => {
```

Comment:
Minor nit: can we actually use [test.for](https://vitest.dev/api/#test-for) with a TestContext?

```suggestion
test.for(['usage.test.ts', 'types.test.ts'])('%s: toMatchTypeOf matches toExtend behaviour', (file, { expect }) => {
```

Reply:
Ah, didn't know about that. I guess we could but I'd rather not just because it's less familiar, to me at least - it relies knowing more about how vitest works. (Also I think your suggestion, isn't right, you'd need to do `([file], {expect})` rather than `(file, {expect})`, it looks like from the docs?